src/meta/client/MetaClient.cpp

```diff
@@ -824,5 +843,20 @@ SchemaVer MetaClient::getNewestEdgeVerFromCache(const GraphSpaceID& space,
     return it->second;
 }
 
+folly::Future<StatusOr<bool>> MetaClient::heartbeat() {
+    CHECK_NOTNULL(listener_);
+    auto localHost = listener_->getLocalHost();
+    cpp2::HBReq req;
+    nebula::cpp2::HostAddr thriftHost;
+    thriftHost.set_ip(localHost.first);
+    thriftHost.set_port(localHost.second);
+    req.set_host(std::move(thriftHost));
+    return getResponse(std::move(req), [] (auto client, auto request) {
+        return client->future_heartBeat(request);
+    }, [] (cpp2::HBResp&& resp) -> decltype(auto) {
```

Comment:
FYI. I sincerely suggest not to use `decltype(auto)` if not necessary while `auto` suffices, because you may return a reference to local storage by accident. Consider this:
```cpp
decltype(auto) get() {
    int var = 0;
    int &ref = var;
    return ref;
    // or return (num);
    // or return (num &= 0xF)
}
```

Reply:
You are right.  We should be careful about it.